### PR TITLE
Add ability to handle vscode.openFolder command

### PR DESCRIPTION
**Pull request by tomer-epstein**
Signed-off-by: Tomer Epstein <tomer.epstein@sap.com>

Issue: https://github.com/eclipse-theia/theia/issues/4050

#### What it does
This change proposal adds ability to handle vscode.openFolder command.

#### How to test

1.  git clone https://github.com/tomer-epstein/vscode-open-folder.git
2. cd vscode-open-folder
3. npm run compile && npm run pacage
4. copy vscode-open-folder-0.0.1.vsix to plugins folder
5. Right click on a folder in the Explorer, choose 'Open Folder New Instance' /  'Open Folder Here' or 'Open Folder Dialog'.

image

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts [hunk 1]
```diff
@@ -101,6 +108,20 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
             }
         });
 
+        commands.registerCommand(VscodeCommands.OPEN_FOLDER, {
+            isVisible: () => false,
+            execute: async (resource?: URI, options?: WorkspaceInput | undefined) => {
```
Comment (akosyakov) [inline packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts]: I'm not suer about `WorkspaceInput`.

We should use VS Code type definition for options and translate it to `WorkspaceInput`. Could you look up the relevant code in VS Code?

Comment (tomer-epstein) [reply packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts]: vscode openFolder api - https://github.com/microsoft/vscode/blob/2af91664d482c6065f7a0d589823ec115a65ec10/src/vs/workbench/api/common/apiCommands.ts#L43

File: packages/plugin-ext/src/common/plugin-api-rpc-model.ts [hunk 2]
```diff
@@ -434,6 +434,12 @@ export interface WorkspaceFolder {
     index: number;
 }
 
+export interface IOpenFolderAPICommandOptions {
+    forceNewWindow?: boolean;
+    forceReuseWindow?: boolean;
+    noRecentEntry?: boolean;
+}
+
```
Comment (tomer-epstein) [inline packages/plugin-ext/src/common/plugin-api-rpc-model.ts]: @akosyakov is this is the right place for this interface?

Comment (akosyakov) [reply packages/plugin-ext/src/common/plugin-api-rpc-model.ts]: I think we can keep it in the vscode command contribution file for now, it is not used anywhere else

Comment (tomer-epstein) [reply packages/plugin-ext/src/common/plugin-api-rpc-model.ts]: Done

**Comment by amiramw** [comment]
@akosyakov i'm confused with regard to git pod workspace created in this PR. When I login and open chrome dev tools I don't see the changes. Is there something wrong with the CI or I am missing something?

**Comment by tomer-epstein** [comment]
@amiramw , try this:

image

**Comment by akosyakov** [comment]
@amiramw When you log in you see Gitpod Theia, not Theia from the PR. There should be a notification to open a Theia from the PR, or you can open it from the port view as @tomer-epstein pointed out

**Comment by akosyakov** [comment]
@amiramw Could you finish the review? If it works lease merge, code wise it looks good to me.

**Comment by amiramw** [comment]
Tested successfully and code looks good to me.

**Comment by akosyakov** [comment]
@amiramw @tomer-epstein please check the git history next time, there are 3 commits in this PR for one file change :( https://github.com/eclipse-theia/theia/blob/master/doc/pull-requests.md#checklist-commit-history

**Comment by tomer-epstein** [comment]
@akosyakov can you pls update https://github.com/eclipse-theia/theia/issues/4050
[ ]  vscode.openFolder - Open a folder or workspace in the current window or new window

**Comment by amiramw** [comment]
> @amiramw @tomer-epstein please check the git history next time, there are 3 commits in this PR for one file change :( https://github.com/eclipse-theia/theia/blob/master/doc/pull-requests.md#checklist-commit-history

Sorry. I missed that.

**Pull request merged**